rllib/core/learner/learner.py

```diff
@@ -266,69 +266,85 @@ class Learner:
 
     Usage pattern:
 
-        .. code-block:: python
-
-        # create a single agent RL module spec.
-        module_spec = SingleAgentRLModuleSpec(
-            module_class=MyModule,
-            observation_space=env.observation_space,
-            action_space=env.action_space,
-            model_config_dict = {"hidden": [128, 128]}
-        )
+        .. testcode::
 
-        # create a learner instance that will train the module
-        learner = MyLearner(module_spec=module_spec)
+            # We use PPO and torch as an example here because many of the showcased
+            # components need implementations to come together. However, the same
+            # pattern is generally applicable.
 
-        # Note: the learner should be built before it can be used.
-        learner.build()
+            from ray.rllib.algorithms.ppo.torch.ppo_torch_rl_module import (
+                PPOTorchRLModule
+            )
+            from ray.rllib.algorithms.ppo.ppo_catalog import PPOCatalog
+            from ray.rllib.core.learner.torch.torch_learner import TorchLearner
+            import gymnasium as gym
 
-        # take one gradient update on the module and report the results
-        results = learner.update(batch)
+            env = gym.make("CartPole-v1")
 
-        # add a new module, perhaps for league based training
-        learner.add_module(
-            module_id="new_player",
-            module_spec=SingleAgentRLModuleSpec(
-                module_class=NewPlayerModule,
+            # Create a single agent RL module spec.
+            module_spec = SingleAgentRLModuleSpec(
+                module_class=PPOTorchRLModule,
                 observation_space=env.observation_space,
                 action_space=env.action_space,
-                model_config_dict = {"hidden": [128, 128]}
+                model_config_dict = {"hidden": [128, 128]},
+                catalog_class = PPOCatalog,
             )
-        )
 
-        # Take another gradient update with both previous and new modules.
-        results = learner.update(batch)
+            # Create a learner instance that will train the module
+            learner = TorchLearner(module_spec=module_spec)
+
+            # Note: the learner should be built before it can be used.
+            learner.build()
 
-        # remove a module
-        learner.remove_module("new_player")
+            # Take one gradient update on the module and report the results
+            # results = learner.update(...)
```

Comment:
(Here and below) What would happen if we uncommented this and replaced the ellipses with an actual argument?

Reply:
We would need to construct a multi-agent training batch here.
This would be a nested dict with quite a lot of KV pairs with all values being torch tensors.
It's just too bug of a thing to construct here. We need to be able to generate example batches some time in the future but we are not there today.